patches/server/0966-Call-BlockGrowEvent-for-missing-blocks.patch

```diff
@@ -0,0 +1,176 @@
+             i = Math.min(i + 1, 25);
+         }
+diff --git a/src/main/java/net/minecraft/world/level/block/MangrovePropaguleBlock.java b/src/main/java/net/minecraft/world/level/block/MangrovePropaguleBlock.java
+index 44c78f0d56c3459c063c104e401a521e3df7d8e5..69df3ac280c539f761e73049fb529a94d5b095d8 100644
+--- a/src/main/java/net/minecraft/world/level/block/MangrovePropaguleBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/MangrovePropaguleBlock.java
+@@ -95,7 +95,7 @@ public class MangrovePropaguleBlock extends SaplingBlock implements SimpleWaterl
+ 
+         } else {
+             if (!isFullyGrown(state)) {
+-                world.setBlock(pos, state.cycle(AGE), 2);
++                org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(world, pos, state.cycle(AGE), 2); // Paper
+             }
+ 
+         }
+@@ -114,7 +114,7 @@ public class MangrovePropaguleBlock extends SaplingBlock implements SimpleWaterl
+     @Override
+     public void performBonemeal(ServerLevel world, RandomSource random, BlockPos pos, BlockState state) {
+         if (isHanging(state) && !isFullyGrown(state)) {
+-            world.setBlock(pos, state.cycle(AGE), 2);
++            org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(world, pos, state.cycle(AGE), 2); // Paper
+         } else {
+             super.performBonemeal(world, random, pos, state);
+         }
+diff --git a/src/main/java/net/minecraft/world/level/block/PitcherCropBlock.java b/src/main/java/net/minecraft/world/level/block/PitcherCropBlock.java
+index 849c0ae6ca30691134f51e5d4d5da372e0e728bd..e9f85f559710da6a9c91e1e149103a307479dfd2 100644
+--- a/src/main/java/net/minecraft/world/level/block/PitcherCropBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/PitcherCropBlock.java
+@@ -126,11 +126,16 @@ public class PitcherCropBlock extends DoublePlantBlock implements BonemealableBl
+     private void grow(ServerLevel world, BlockState state, BlockPos pos, int amount) {
+         int i = Math.min(state.getValue(AGE) + amount, 4);
+         if (this.canGrow(world, pos, state, i)) {
+-            world.setBlock(pos, state.setValue(AGE, Integer.valueOf(i)), 2);
+             if (i >= 3) {
+                 BlockPos blockPos = pos.above();
++                if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(world, blockPos, pos, copyWaterloggedFrom(world, pos, this.defaultBlockState().setValue(AGE, Integer.valueOf(i)).setValue(HALF, DoubleBlockHalf.UPPER)), 2)) return; // Paper
+                 world.setBlock(blockPos, copyWaterloggedFrom(world, pos, this.defaultBlockState().setValue(AGE, Integer.valueOf(i)).setValue(HALF, DoubleBlockHalf.UPPER)), 3);
+             }
++            // Paper start
++            else {
++                org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(world, pos, state.setValue(AGE, Integer.valueOf(i)), 2);
++            }
++            // Paper end
```

Comment:
Just confirming that removing the double setBlock isn't gonna break any obscure mechanic right? something that relies on updating something somewhere maybe?

Reply:
Like in vanilla, if `i >= 3`, there will be 2 consecutive setBlock calls right? the one being deleted here, and the one below the first grow event handler. Removing one of them isn't going to break anything right?

Reply:
the event call is the setblock